### PR TITLE
Support maps in BidiEngine (t18589)

### PR DESCRIPTION
**Pull request by semion1956**

### CODE REVIEW

File: string/tests/BidiEngine/BidiEngineMapsTest.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+dojo.provide("dojox.string.tests.BidiEngine.BidiEngineMapsTest");
```
Comment (dylans) [inline string/tests/BidiEngine/BidiEngineMapsTest.js]: Any new code/tests need to use AMD syntax please.

**Comment by semion1956** [comment]
@ddumontatibm  FYI

**Comment by cjolif** [comment]
Maybe you need to make sure to not changes spacing in BidiEngine.js to ease looking at the diff.

**Comment by ddumontatibm** [comment]
@cjolif adding ?w=1 to the file view helps a bit.   But I would recommend against changing any whitespace conventions in use in the current project or any whitespace that currently exists in the project to ease the review process.

**Comment by semion1956** [comment]
@dylans  AMD syntax added.

**Comment by ddumontatibm** [comment]
I think you may have an issue with the git setup you are using.  
Are you using windows and eclipse + egit?

**Comment by semion1956** [comment]
Yes, I use windows and eclipse. What is the issue, that you wrote about?

**Comment by ddumontatibm** [comment]
Egit has long outstanding bugs for ignoring line ending settings.  I'll take another look to see if that's the issue, but I did not expect the diff of this amd wrapper to be so severe.   Perhaps the GH diff viewer is just not that great.

**Comment by cjolif** [comment]
ok, it seems the spacing issue was actually in the initial file and the new commit is fixing those. Otherwise looks good to me. So I'll be merging that soon if there are no further comments.

**Comment by cjolif** [comment]
merged in 0abf3b264cf3567305b7a7e31619bb65ae195104

**Pull request closed without merging**